server/service/src/sensor/berlinger.rs

```diff
@@ -0,0 +1,380 @@
+            id: uuid(),
+            store_id: sensor_row.store_id.clone(),
+            is_active: true,
+            description: config_description.clone(),
+            duration: temperature_breach_config.duration.num_seconds() as i32,
+            r#type: breach_row_type,
+            minimum_temperature: temperature_breach_config.minimum_temperature,
+            maximum_temperature: temperature_breach_config.maximum_temperature,
+        };
+        TemperatureBreachConfigRowRepository::new(connection)
+            .upsert_one(&new_temperature_breach_config)?;
+        println!(
+            "Added sensor breach config {:?} ",
+            new_temperature_breach_config
+        );
+        Ok(())
+    }
+}
+
+fn sensor_add_if_new(
+    connection: &StorageConnection,
+    store_id: &str,
+    temperature_sensor: &temperature_sensor::Sensor,
+) -> Result<(), RepositoryError> {
+    let result = get_matching_sensor_serial(connection, &temperature_sensor.serial)?;
+
+    if let Some(_record) = result.clone().pop() {
+        Ok(())
+    } else {
+        let mut interval_seconds = None;
+        if let Some(interval_duration) = temperature_sensor.log_interval {
+            interval_seconds = Some(interval_duration.num_seconds() as i32);
+        }
+        let new_sensor = SensorRow {
+            id: uuid(),
+            serial: temperature_sensor.serial.clone(),
+            name: temperature_sensor.name.clone(),
+            store_id: Some(store_id.to_string()),
+            location_id: None,
+            last_connection_datetime: None,
+            battery_level: None,
+            is_active: true,
+            log_interval: interval_seconds,
+            r#type: SensorType::Berlinger,
+        };
+        SensorRowRepository::new(connection).upsert_one(&new_sensor)?;
+        println!("Added sensor {:?} ", new_sensor);
+        Ok(())
+    }
+}
+
+pub fn read_sensors(
+    connection: &StorageConnection,
+    store_id: &str,
+) -> Result<Vec<String>, SingleRecordError> {
+    let mut sensors_processed: Vec<String> = Vec::new();
+
+    let sensor_serials =
+        read_connected_serials().map_err(|err| SingleRecordError::NotFound(err))?;
+    let expected_sensor_count = sensor_serials.len();
+
+    for current_serial in sensor_serials {
+        let mut temperature_sensor = temperature_sensor::read_sensor(&current_serial, true)
+            .map_err(|err| SingleRecordError::NotFound(err))?;
+        sensor_add_if_new(connection, &store_id, &temperature_sensor)?;
+
+        let result = get_matching_sensor_serial(connection, &current_serial)?;
+
+        if let Some(mut record) = result.clone().pop() {
+            let sensor_id = &record.sensor_row.id;
+            // Filter sensor data by previous last connected time
+            let last_connected = record.sensor_row.last_connection_datetime;
+            temperature_sensor =
+                temperature_sensor::filter_sensor(temperature_sensor, last_connected, None, true);
```

Comment:
Does that filter logs from sensor ?

If so, I think a more explicit name would be usefull ? filter_sensor_logs_and_breachs ?

Reply:
It filters everything (currently that's logs and breaches) - not sure that any extra wording is needed as it doesn't really add any useful information IMHO?

Reply:
Ok, would be good to have a small comment at the consumer end about this, makes reading the top level logic easier. I made the comment because it wasn't clear on the surface.